### DISCUSSION TITLE
Does this leverage any kind of persistent memory?

### DISCUSSION THREAD

**Discussion by interior-night** [intitni/CopilotForXcode#265] [General]
I'm wondering how this compares to something like https://github.com/AntonOsika/gpt-engineer? Does CoPilot/ChatGPT remember previous prompts and take them into account (for example, if i tell it to never use UIKit)?

**Comment by intitni** [comment]
apt-engineer works more like an agent, it may have implemented some memory retrieval solution but I don't know. Usually ChatGPT only remembers the latest few messages and the system prompt. To let it remember to never use UIKit, you can put the instruction inside system prompt. 

In Copilot for Xcode, you can 
- change the default system prompt in the host app
- use custom chat command to override the system prompt

or temporarily change it by
- right-clicking a message and set the message as extra system prompt
- use custom open chat command to override the extra system prompt
